modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/LoggerContextConfigurer.java

```diff
@@ -179,20 +184,21 @@ private void addDefaultArtifactContext(MuleLoggerContext context) {
     File logFile = new File(logDir, logName);
 
     if (context.getConfigLocation() == null) {
-
       addDefaultAppender(context, logFile.getAbsolutePath());
-    } else {
+    } else if (isUrlInsideDirectory(context.getConfigFile(), MuleContainerBootstrapUtils.getMuleConfDir())
+        && !hasFileAppender(context)
+        || context.getConfiguration().getAppenders().isEmpty()) {
```

Comment:
for performance reasons, maybe the inside directory check can be made at the end (move to the right) since it goes to the filesystem.